Some stubbed Vuetify components can't be found with `find` wrapper method

### Version
1.0.0-beta.29

### Reproduction link
[https://github.com/anthonygore/vuetify-test](https://github.com/anthonygore/vuetify-test)







### Steps to reproduce
npm i
npm run test:unit

### What is expected?
All tests should pass

### What is actually happening?
The second test fails as the VTextField component can't be found.

<!-- generated by vue-issues. DO NOT REMOVE -->

* * * 

I don't know if this is a Vuetify issue or a Vue Test Utils issue, but some stubbed child components can't be found with the `find` wrapper method when the parent is shallow mounted.

It's not just `VTextField`, either, `VSelect` also has this problem, while others like `VIcon` work just fine.


```js
import { shallowMount } from "@vue/test-utils";
import Vuetify from "vuetify";
import { VBtn, VTextField } from "vuetify/lib";

Vue.use(Vuetify);

const component = {
  template: `
    <div>
      <v-btn />
      <v-text-field />
    </div>
  `
};

// Passes
it('should find v-btn', () => {
  const wrapper = shallowMount(component);
  expect(wrapper.find(VBtn).exists()).toBe(true);
});

// Fails
it('should find v-text-field', () => {
  const wrapper = shallowMount(component);
  expect(wrapper.find(VTextField).exists()).toBe(true);
});
```

### Update

I think the issue is something to do with how some global components are registered because this works:

```js
Vue.use(Vuetify);

const component = {
  template: `
    <div>
      <v-btn />
      <v-text-field />
    </div>
  `,
  components: {
    VTextField
  }
};
```

Note that I don't need to locally register `VBtn`, just `VTextField`...

> Are you sure? `v-icon` [is functional](https://github.com/vuetifyjs/vuetify/blob/6f32c6439a4c789ac992794f5d514db6fec4a5fc/packages/vuetify/src/components/VIcon/VIcon.ts#L147) and I can `find` that one.
> 
> ```js
> import { VIcon } from "vuetify/lib";
> 
> it('should find v-icon', () => {
>   const wrapper = shallowMount(component);
>   expect(wrapper.find(VIcon).exists()).toBe(true); // Passes
> });
> ```

For me, I cant use `wrapper.find` with a VBtn, for example. The console returns:
```
 TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
    tests/unit/infosets/components/InfosetsHeader.spec.ts:17:25 - error TS2345: Argument of type 'Component<DefaultData<never>, DefaultMethods<never>, DefaultComputed, Record<string, any>>' is not assignable to parameter of type 'NameSelector'.
      Type 'FunctionalComponentOptions<Record<string, any>, PropsDefinition<Record<string, any>>>' is not assignable to type 'NameSelector'.
        Types of property 'name' are incompatible.
          Type 'string | undefined' is not assignable to type 'string'.
            Type 'undefined' is not assignable to type 'string'.

    17     expect(wrapper.find(VBtn).exists()).toBe(true);
```

--- a/packages/create-instance/create-instance.js
+++ b/packages/create-instance/create-instance.js
@@ -52,8 +52,14 @@ export default function createInstance(
   // root instance when it's instantiated
   const instanceOptions = extractInstanceOptions(options)
 
+  const globalComponents = _Vue.options.components || {}
+  const componentsToStub = Object.assign(
+    Object.create(globalComponents),
+    componentOptions.components
+  )
+
   const stubComponentsObject = createStubsFromStubsObject(
-    componentOptions.components,
+    componentsToStub,
     // $FlowIgnore
     options.stubs,
     _Vue

--- a/test/specs/mounting-options/stubs.spec.js
+++ b/test/specs/mounting-options/stubs.spec.js
@@ -568,4 +568,26 @@ describeWithShallowAndMount('options.stub', mountingMethod => {
     expect(wrapper.find(ToStub).exists()).to.be.false
     expect(wrapper.find(Stub).exists()).to.be.true
   })
+
+  it('stubs globally registered components', () => {
+    const ChildComponent = {
+      template: '<div />',
+      props: ['propA'],
+      name: 'child-component'
+    }
+    const TestComponent = {
+      template: '<child-component prop-a="A" />'
+    }
+
+    Vue.component('child-component', ChildComponent)
+    const wrapper = mountingMethod(TestComponent, {
+      stubs: {
+        ChildComponent: true
+      }
+    })
+    const result = wrapper.find(ChildComponent)
+    expect(result.exists()).to.be.true
+    expect(result.props().propA).to.equal('A')
+    delete Vue.options.components['child-component']
+  })
 })

